### ISSUE TITLE
Sidebar menu changes active menu item too quickly

### ISSUE THREAD

**Issue by SaleRise** [PanJiaChen/vue-element-admin#1190]
If you have **beforeRouteLeave** in the component to do route based check like this:

```js
beforeRouteLeave(to, from, next) {
    this.$confirm('Do you want to proceed?')
      .then(() => {
        next()
      })
      .catch(() => {
        next(false)
      })
  },
```

On cancel, user stays on the page which is expected behavious but the sidebar menu is active. Can this be addressed. Screen capture is attached from reference.

Thanks
![2018-10-12_10-41-52](https://user-images.githubusercontent.com/7464668/46841473-82d8ab80-ce0b-11e8-83cc-c200947de80c.gif)

**Comment by PanJiaChen** [comment]
You have two solutions:
 - Use el-menu`s router 
 This program requires you to modify the source code of sidebar.js

- Change activeIndex
 
```js
beforeRouteLeave(to, from, next) {
    const answer = window.confirm('Do you really want to leave? you have unsaved changes!')
    if (answer) {
      next()
    } else {
      this.$parent.$parent.$children[0].$children[0].$children[0].activeIndex = from.path
      next(false)
    }
  }
```

**Comment by SaleRise** [comment]
Thanks @PanJiaChen for your help. Doesn't it look like a bandage rather than a proper solution. I mean, why has the menu activeIndex update before router took user to that page. It makes sense what was being done but seems like this use-case where pop-up appears, next page is automatically triggered and not looking at vue router.

**Comment by PanJiaChen** [comment]
This is mainly a problem with element-ui.

https://github.com/PanJiaChen/vue-element-admin/blob/master/src/views/layout/components/Sidebar/index.vue

If you want a universal solution, you need to modify the code. Use el-menu`s router.

![image](https://user-images.githubusercontent.com/8121621/46925672-b744a580-d05f-11e8-87ad-d4c05cf42f87.png)

**Comment by PanJiaChen** [comment]
In fact, you don't use el-menu`s router, it's also no way.

**Comment by SaleRise** [comment]
@PanJiaChen Yes I tried that as well. That did not work. Seems like this needs to be reported to Element-UI.

For now I am applying the resolution you suggested in your first post.

Cheers
